src/main/java/org/terasology/additionalRails/action/RailSwitchAction.java

```diff
@@ -0,0 +1,135 @@
+
+    private Block railSwitchLeverOff;
+    private Block railSwitchLeverOn;
+    private Block railSwitchSignalOff;
+    private Block railSwitchSignalOn;
+    private RailsUpdateFamily railFamily;
+    private RailsUpdateFamily invertFamily;
+
+    /**
+     * Prepares blocks for future use (see code below)
+     * These block are obtained from an instance of {@link org.terasology.world.block.BlockManager}
+     */
+    @Override
+    public void initialise() {
+        final BlockManager blockManager = CoreRegistry.get(BlockManager.class);
+        railSwitchLeverOff = blockManager.getBlock("AdditionalRails:railSwitchLeverOff");
+        railSwitchLeverOn = blockManager.getBlock("AdditionalRails:railSwitchLeverOn");
+        railSwitchSignalOff = blockManager.getBlock("AdditionalRails:railSwitchSignalOff");
+        railSwitchSignalOn = blockManager.getBlock("AdditionalRails:railSwitchSignalOn");
+        railFamily = (RailsUpdateFamily)blockManager.getBlockFamily("Rails:rails");
+        invertFamily = (RailsUpdateFamily)blockManager.getBlockFamily("Rails:railsTBlockInverted");
+    }
+
+    /**
+     * Updates railSwitchLever and rails around it, when user interacts the block.
+     * @param event ActivateEvent, just used to differentiate from other events
+     * @param entity The entity related to railSwitchLever
+     */
+    @ReceiveEvent(components = {BlockComponent.class, RailSwitchLeverComponent.class})
+    public void railSwitchLeverAction(ActivateEvent event, EntityRef entity) {
+        RailSwitchLeverComponent rslComponent = entity.getComponent(RailSwitchLeverComponent.class);
+        Vector3i blockLocation = new Vector3i(entity.getComponent(BlockComponent.class).getPosition());
+
+        //switch the block from on to off, from off to on; rslComponent.isOn is set in block prefab
+        if (rslComponent.isOn) {
+            worldProvider.setBlock(blockLocation, railSwitchLeverOff);
+        } else {
+            worldProvider.setBlock(blockLocation, railSwitchLeverOn);
+        }
+
+        //update rails based on switch state
+        updateRails(blockLocation, rslComponent.isOn);
+    }
+
+    /**
+     * Updates railSwitchSignal and rails around it, when {@link org.terasology.signalling} signal changes its state
+     * @param event OnChangedComponent event raised by signal modyfying {@link org.terasology.signalling.components.SignalConsumerStatusComponent}
+     * @param entity The entity related to railSwitchSignal
+     */
+    @ReceiveEvent(components = {BlockComponent.class, RailSwitchSignalComponent.class, SignalConsumerStatusComponent.class})
+    public void railSwitchSignalAction(OnChangedComponent event, EntityRef entity) {
+        SignalConsumerStatusComponent scsComponent = entity.getComponent(SignalConsumerStatusComponent.class);
+        Vector3i blockLocation = new Vector3i(entity.getComponent(BlockComponent.class).getPosition());
+        Block block = worldProvider.getBlock(blockLocation);
+
```

Comment:
Again, you can add the components to the method signature instead of the annotation